config/m4/module.m4

```diff
@@ -0,0 +1,484 @@
+#
+# Copyright (C) 2022 NVIDIA CORPORATION & AFFILIATES. All Rights Reserved.
+# See file LICENSE for terms.
+#
+
+# helper macro
+m4_define([_CHECK_MODULE_MAKE_HELP_STRING], [m4_argn($1, $2)$3])
+    
+m4_ifndef([PKG_CHECK_MODULES_STATIC],
+          [AC_DEFUN([PKG_CHECK_MODULES_STATIC],
+                    [AC_REQUIRE([PKG_PROG_PKG_CONFIG])
+                     save_PKG_CONFIG=$PKG_CONFIG
+                     PKG_CONFIG="$PKG_CONFIG --static"
+                     PKG_CHECK_MODULES([$1], [$2], [$3], [$4])
+                     PKG_CONFIG=$save_PKG_CONFIG
+                    ])])
+
+m4_define([_CHECK_MODULE_DEFVAR],
+          [
+             m4_pushdef([mod_header], [m4_argn([1], $1)])
+             m4_pushdef([mod_decls], [m4_argn([2], $1)])
+             m4_pushdef([mod_includes], [m4_argn([3], $1)])
+             m4_pushdef([mod_libs], [m4_argn([4], $1)])
+             m4_pushdef([mod_funcs], [m4_argn([5], $1)])
+             m4_pushdef([mod_ldflags], [m4_argn([6], $1)])
+          ])
+
+m4_define([_CHECK_MODULE_UNDEFVAR],
+          [
+             m4_popdef([mod_ldflags])dnl
+             m4_popdef([mod_funcs])dnl
+             m4_popdef([mod_libs])dnl
+             m4_popdef([mod_includes])dnl
+             m4_popdef([mod_decls])dnl
+             m4_popdef([mod_header])dnl
+          ])
+
+#
+# Init progress
+#
+# Usage: _CHECK_MODULE_INIT_PROGRESS([mod-dir], [progress],
+#                                    [dir-int], [inc-flag], [dir-lib], [dir-flag], [if-failed])
+#
+m4_define([_CHECK_MODULE_INIT_PROGRESS],
+          [
+            _CHECK_MODULE_IS_YNG([$1],
+                                 [$2=yes],   # yes
+                                 [$2=skip],  # no
+                                 [$2=guess], # guess
+                                 [$2=skip],  # empty
+                                 [$2=dir     # DIR
+                                  _CHECK_MODULE_SET_DIR([$1], [$3], [$4], [$5],
+                                                        [$6], [$7])])
+          ])
+
+#
+# Configure module directories
+#
+# Usage: _CHECK_MODULE_SET_DIR([mod-dir], [inc], [incflag], [lib], [libflag], [if-failed])
+#
+m4_define([_CHECK_MODULE_SET_DIR],
+          [
+             $2=$1/include
+             $3=-I$1/include
+             # use macro unwind
+             m4_pushdef([dir_lib], [$4])
+             _CHECK_MODULE_FIXUP_LIB([$1], [$4], [$5=-L$dir_lib], [$6])
+             m4_popdef([dir_lib])
+          ])
+
+#
+# Run test if status is in allowed list
+#
+# Usage: _CHECK_MODULE_RUN([status], [allowed-statuses], [macro], [if-no-macro])
```

Comment:
name and doc not clear..
did to mean to check if $1 is equal to any of the values listed in $2 ?